galsim/fits.py

```diff
@@ -1119,6 +1119,7 @@ def __init__(self, header=None, file_name=None, dir=None, hdu_list=None, hdu=Non
             if header is not None:
                 if hasattr(header, 'items'):
                     # update() should handle anything that acts like a dict.
+                    header = {k.upper(): v for k, v in header.items()}
```

Comment:
You may have trouble in python 2 with this line...

Reply:
It's fine in 2.7.  It was only invalid in 2.6, which we don't support anymore.

The test failures are something weird about how PyAst is pickling ZPX WCS's.  It apparently works fine in 3.6, but not 2.7 or 3.5.  I'll figure it out.

Reply:
Ahhh good to know.